data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/LogicalReplicationEventProcessor.java

```diff
@@ -432,13 +435,16 @@ private String getNullTerminatedString(ByteBuffer msg) {
         return sb.toString();
     }
 
-    private List<String> getPrimaryKeys(String schemaName, String tableName) {
-        final String databaseName = sourceConfig.getTableNames().get(0).split("\\.")[0];
+    private List<String> getPrimaryKeys(String databaseName, String schemaName, String tableName) {
         StreamProgressState progressState = streamPartition.getProgressState().get();
 
         return progressState.getPrimaryKeyMap().get(databaseName + "." + schemaName + "." + tableName);
     }
 
+    private String getDatabaseName(List<String> tableNames) {
+        return tableNames.get(0).split("\\.")[0];
```

Comment:
nit: move the split delimiter to constant

Reply:
Done.